substratevm/src/com.oracle.svm.hosted/src/com/oracle/svm/hosted/phases/IntrinsifyMethodHandlesInvocationPlugin.java

```diff
@@ -487,9 +492,14 @@ private void processInvokeWithMethodHandle(GraphBuilderContext b, Replacements r
              */
             Node singleFunctionality = null;
             ReturnNode singleReturn = null;
+            ValueNode singleNewInstance = null;
             for (Node node : graph.getNodes()) {
                 if (node == graph.start() || node instanceof ParameterNode || node instanceof ConstantNode || node instanceof FrameState) {
                     /* Ignore the allowed framework around the nodes we care about. */
+                } else if (node instanceof DynamicNewInstanceNode && singleNewInstance == null && ((DynamicNewInstanceNode) node).getInstanceType().isConstant()) {
```

Comment:
I just merged https://github.com/oracle/graal/commit/18bdfddcbedf05331a47d628f0b1f4be60a52985 which ensures that `DynamicNewInstanceNode` with a constant class is always constant folded to a `NewInstanceNode`. Please remove this check now (and everything else related to `DynamicNewInstanceNode` in this PR), it should no longer be necessary.